collect_app/src/main/java/org/odk/collect/android/upload/InstanceGoogleSheetsUploader.java

```diff
@@ -338,20 +338,26 @@ private void createSheetsIfNeeded(TreeElement element) throws UploadException {
         Set<String> sheetTitles = new HashSet<>();
         for (TreeElement childElement : getChildElements(element, false)) {
             if (childElement.isRepeatable()) {
-                sheetTitles.add(getElementTitle(childElement));
+                sheetTitles.add(shortenSheetTitleIfTooLong(getElementTitle(childElement)));
                 sheetTitles.addAll(getSheetTitles(childElement));
             }
         }
         return sheetTitles;
     }
 
+    private String shortenSheetTitleIfTooLong(String sheetTitle) {
+        return sheetTitle.length() <= 100
+                ? sheetTitle
+                : "..." + sheetTitle.substring(sheetTitle.length() - 97, sheetTitle.length());
+    }
```

Comment:
Can we make this a more generic TextUtils method that takes a string and a length (e.g. `ellipsizeBeginning`)? This would also be really easy to unit test so I think we should do that.